File: app/src/main/java/com/rowland/delivery/features/dash/domain/usecases/product/CreateProductUseCase.java
package com.rowland.delivery.features.dash.domain.usecases.product;

import com.rowland.delivery.features.dash.data.model.product.ProductRepository;
import com.rowland.delivery.features.dash.domain.models.category.Category;
import com.rowland.delivery.features.dash.domain.models.product.Product;

import javax.inject.Inject;

import io.reactivex.Single;

/**
 * Created by Rowland on 5/13/2018.
 */

public class CreateProductUseCase {

    private ProductRepository productRepository;

    @Inject
    public CreateProductUseCase(ProductRepository productRepository) {
        this.productRepository = productRepository;
    }

    public Single<Product> createProduct(Product product, String userUid) {
        return this.productRepository.createPoduct(product, userUid);
    }
}

File: app/src/main/java/com/rowland/delivery/features/dash/domain/usecases/product/CreateProductUseCase.kt
package com.rowland.delivery.features.dash.domain.usecases.product

import com.rowland.delivery.features.dash.domain.contracts.IProductRepository
import com.rowland.delivery.features.dash.domain.models.product.ProductEntity
import io.reactivex.Single
import javax.inject.Inject

/**
 * Created by Rowland on 5/13/2018.
 */

class CreateProductUseCase @Inject
constructor(private val productRepository: IProductRepository) {

    fun createProduct(productEntity: ProductEntity, productCategory: String, userUid: String): Single<ProductEntity> {
        return this.productRepository.createProduct(productEntity, userUid, productCategory)
    }
}
